src/code/views/share-dialog-tabs-view.tsx

```diff
@@ -0,0 +1,208 @@
+      </div>
+      <div className='social-icons'>
+        <SocialIcon icon='facebook' url={facebookUrl} />
+        <SocialIcon icon='twitter' url={twitterUrl} />
+      </div>
+    </div>
+  )
+}
+
+export type ShareDialogTab = 'api' | 'embed' | 'lara' | 'link'
+export interface IShareDialogLaraApiTabProps {
+  linkUrl: string;
+  serverUrlLabel: string;
+  serverUrl: string;
+  onChangeServerUrl: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+interface IShareDialogTabsProps {
+  tabSelected: ShareDialogTab;
+  linkUrl: string;
+  embedUrl: string;
+  lara?: IShareDialogLaraApiTabProps;
+  interactiveApi?: IShareDialogLaraApiTabProps;
+  fullscreenScaling?: boolean;
+  visibilityToggles?: boolean;
+  onChangeFullscreenScaling?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onChangeVisibilityToggles?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onSelectTab: (tab: ShareDialogTab) => void;
+  onCopyClick: (e: React.MouseEvent) => void;
+}
+export const ShareDialogTabsView: React.FC<IShareDialogTabsProps> = ({
+              tabSelected, embedUrl, linkUrl, lara, interactiveApi, onSelectTab, onCopyClick, ...others
+}) => {
+  return (
+    <div data-testid='share-dialog-tabs-view'>
+      <ul className='sharing-tabs'>
+        <li className={classNames('sharing-tab', 'sharing-tab-link', { 'sharing-tab-selected': tabSelected === 'link' })}
+            style={{ marginLeft: 10 }} onClick={() => onSelectTab('link')} data-testid='sharing-tab-link'>
+          {translate("~SHARE_DIALOG.LINK_TAB")}
+        </li>
+        <li className={classNames('sharing-tab', 'sharing-tab-embed', { 'sharing-tab-selected': tabSelected === 'embed' })}
+            onClick={() => onSelectTab('embed')} data-testid='sharing-tab-embed'>
+          {translate("~SHARE_DIALOG.EMBED_TAB")}
+        </li>
+        {lara &&
+          <li className={classNames('sharing-tab', 'sharing-tab-lara', { 'sharing-tab-selected': tabSelected === 'lara' })}
+              onClick={() => onSelectTab('lara')} data-testid='sharing-tab-lara'>
+            LARA
+          </li>}
+        {interactiveApi &&
+          <li className={classNames('sharing-tab', 'sharing-tab-api', { 'sharing-tab-selected': tabSelected === 'api' })}
+              onClick={() => onSelectTab('api')} data-testid='sharing-tab-api'>
+            Interactive API
+          </li>}
+      </ul>
+      <div className="sharing-tab-contents">
+        {(() => {
+          switch (tabSelected) {
+            case 'link':
+              return <LinkTabContents url={linkUrl} onCopyClick={onCopyClick} />
+            case 'embed':
+              return <EmbedTabContents url={embedUrl} onCopyClick={onCopyClick} />
+            case 'lara': {
+              const { linkUrl: url, ...laraOthers } = lara
+              return <LaraApiTabContents mode='lara' linkUrl={url} onCopyClick={onCopyClick}
+                                        {...others} {...laraOthers} />
+            }
```

Comment:
Seems more consistent to use `{..}` in each case if you have to use it here.

Reply:
The `link` and `embed` tabs only need two props. All the others are for the use of the `lara` and `api` tabs.